### PR TITLE
Add _circuit_diagram_info_ to cirq.PauliString

### PR DESCRIPTION
**Pull request by Strilanc**
- Also adds `CircuitDiagramInfoArgs.format_complex`

```
0: ───PauliString(+X)───PauliString(-X)───PauliString(+X)───PauliString(iX)───
                                          │                 │
1: ───────────────────────────────────────┼─────────────────Y─────────────────
                                          │
2: ───────────────────────────────────────Z───────────────────────────────────
```

### CODE REVIEW

File: cirq/ops/pauli_string.py [hunk 1]
```diff
@@ -259,6 +259,20 @@ def keys(self) -> KeysView[raw_types.Qid]:
...
+    def _circuit_diagram_info_(self, args: 'cirq.CircuitDiagramInfoArgs'
+                              ) -> List[str]:
+        if not len(self._qubit_pauli_map):
+            return NotImplemented
+
+        qs = args.known_qubits or list(self._qubit_pauli_map.keys())
+        symbols = list(str(self.get(q)) for q in qs)
+        prefix = ('+' if self.coefficient == +1 else '-' if
+                  self.coefficient == -1 else 'i' if self.coefficient == 1j else
+                  '-i' if self.coefficient == -1j else repr(self.coefficient) +
```
Comment (viathor) [inline cirq/ops/pauli_string.py:271]: The case of PauliStrings outside the Pauli group, i.e. with coefficients not in {1, -1, i, -i} may render a long ugly floating point number. Perhaps add explicit format string, e.g. limiting to three or four decimal places.

File: cirq/ops/pauli_string_test.py [hunk 2]
```diff
@@ -1558,3 +1558,23 @@ def test_deprecated():
...
+
+    assert cirq.circuit_diagram_info(cirq.PauliString(), default=None) is None
+
+    cirq.testing.assert_has_diagram(
+        cirq.Circuit(
+            cirq.PauliString({a: cirq.X}),
+            -cirq.PauliString({a: cirq.X}),
+            cirq.X(a) * cirq.Z(c),
+            1j * cirq.X(a) * cirq.Y(b),
+        ), """
```
Comment (viathor) [inline cirq/ops/pauli_string_test.py:1574]: Cover the case outside the Pauli group, e.g. `cirq.PauliString({a: cirq.X}, coefficient=np.sqrt(1j))`.

Comment (Strilanc) [reply cirq/ops/pauli_string_test.py:1576]: Done.

File: cirq/ops/pauli_string.py (hunk 1 above)
Comment (Strilanc) [reply cirq/ops/pauli_string.py]: Done.

**Pull request merged**